Filters/ProductDiscountPriceFilter.cs

```diff
@@ -0,0 +1,48 @@
+﻿using System;
+using Nwazet.Commerce.Extensions;
+using Nwazet.Commerce.Models;
+using Orchard.ContentManagement;
+using Orchard.Environment;
+using Orchard.Localization;
+using Orchard.Projections.Descriptors.Filter;
+using Orchard.Projections.FilterEditors.Forms;
+using Orchard.Projections.Services;
+using Orchard.UI.Resources;
+
+namespace Nwazet.Commerce.Filters {
+
+    public class ProductDiscountPriceFilter : IFilterProvider {
+        public Localizer T { get; set; }
+        private readonly Work<IResourceManager> _resourceManager;
+        public ProductDiscountPriceFilter( Work<IResourceManager> resourceManager) {
+            _resourceManager = resourceManager;
+
+            T = NullLocalizer.Instance;
+        }
+
+        public void Describe(DescribeFilterContext describe) {
+            describe.For("ProductPart", T("Product"), T("Product"))
+                .Element("ProductDiscountPrice", T("Product DiscountPrice"), T("Product DiscountPrice Filter."),
+                    (Action<dynamic>)ApplyFilter,
+                    (Func<dynamic, LocalizedString>)DisplayFilter,
+                    "ProductDiscountPriceFilterForm"
+                );
+        }
+
+        public void ApplyFilter(dynamic context) {
+            string value = context.State.Value;
+            string min = context.State.Min;
+            string max = context.State.Max;
+            var op = (NumericOperator)Enum.Parse(typeof(NumericOperator), Convert.ToString(context.State.Operator));
```

Comment:
See, that's what I was talking about above: it seems like the parsing belongs here, not in the actual helpers. You are parsing the operator, and it's inconsistent not to parse the values at the same place.